pkg/controller/build/build_controller.go

```diff
@@ -48,8 +48,21 @@ func add(mgr manager.Manager, r reconcile.Reconciler) error {
 
 	pred := predicate.Funcs{
 		UpdateFunc: func(e event.UpdateEvent) bool {
+			buildRunFinalizer := false
```

Comment:
I think you can improve here. First is always better to cast to the object you are expecting, like:
```
o := e.ObjectOld.(*build.Build)
n := e.ObjectNew.(*build.Build)
```
in order to access the Object struct fields, is just much easier.

Reply:
Good idea,
Changed to:
```
			oldAnnot := e.MetaOld.GetAnnotations()
			newAnnot := e.MetaNew.GetAnnotations()
```

Reply:
I dont think this is what I meant.  You need to cast the right object first(a Build), is a best practice, otherwise how you know which object you are accesing?. In the future, we can get access to the whole Build object fields

Reply:
? I think in our controller we just watch the and operate Build object.
We don't need to check the object type first.

Like before, it is just a:
`return e.MetaOld.GetGeneration() != e.MetaNew.GetGeneration()`
We don't watch ALL objects and don't need to check if it is a Build then check if the Generations are different.

We just watch the build resource here:
`err = c.Watch(&source.Kind{Type: &buildv1alpha1.Build{}}`